service/src/main/java/org/sunbird/actor/user/SSOUserCreateActor.java

```diff
@@ -269,22 +271,54 @@ private String findRootOrgId(Request actorMessage) {
   }
 
   private void populateRoles(Request actorMessage, String rootOrgId ) {
-    Map<String, Object> userMap = actorMessage.getRequest();
+    Map<String, Object> userMap = (Map<String, Object>) actorMessage.getRequest();
     if (userMap.get(JsonKey.ROLES) == null || ((List<String>) userMap.get(JsonKey.ROLES)).isEmpty()) {
       userMap.put(JsonKey.ROLES, Arrays.asList(JsonKey.PUBLIC));
     } else {
       checkIfMDOLeaderExist(userMap, actorMessage, rootOrgId);
     }
   }
 
-  private void createBasisProfileDetails(Request actorMessage) {
+  private void createBasisProfileDetails(Request actorMessage) throws JsonProcessingException {
     Map<String, Object> userMap = actorMessage.getRequest();
     Map<String, Object> profileDetails = new HashMap<>();
-    profileDetails.put(JsonKey.PROFILE_GROUP_STATUS, "NOT-VERIFIED");
-    profileDetails.put(JsonKey.PROFILE_DESIGNATION_STATUS, "NOT-VERIFIED");
-    profileDetails.put(JsonKey.PROFILE_STATUS, "NOT-VERIFIED");
-    profileDetails.put(JsonKey.MANDATORY_FIELDS_EXISTS, false);
-    userMap.put(JsonKey.PROFILE_DETAILS, profileDetails.toString());
+    Map<String, Object> employmentDetails = Map.of(JsonKey.DEPARTMENT_NAME, userMap.getOrDefault(JsonKey.CHANNEL, ""));
+    Map<String, Object> additionalProperties = new HashMap<>();
+    List<Map<String, Object>> professionalDetailsList = new ArrayList<>();
+    Map<String, Object> professionalDetails = new HashMap<>();
+
+    Map<String, Object> personalDetailsRequest = (Map<String, Object>) actorMessage.get(JsonKey.PERSONAL_DETAILS);
+    Map<String, Object> personalDetails = new HashMap<>(Map.of(
+            JsonKey.MOBILE, userMap.getOrDefault(JsonKey.PHONE, ""),
+            JsonKey.PRIMARY_EMAIL, userMap.getOrDefault(JsonKey.EMAIL, ""),
+            JsonKey.FIRST_NAME, userMap.getOrDefault(JsonKey.FIRST_NAME, "")
+    ));
+
+    personalDetails.put(JsonKey.ROLES, userMap.getOrDefault(JsonKey.ROLES, new ArrayList<>()));
+
+    if (MapUtils.isNotEmpty(personalDetailsRequest)) {
+      personalDetails.putAll(personalDetailsRequest);
+
+      additionalProperties.put(JsonKey.TAGS, personalDetails.remove(JsonKey.TAGS));
+      professionalDetails.put(JsonKey.DESIGNATION, personalDetails.remove(JsonKey.DESIGNATION));
+      professionalDetails.put(JsonKey.GROUP, personalDetails.remove(JsonKey.GROUP));
+    }
+
+    professionalDetailsList.add(professionalDetails.isEmpty() ?
+            Map.of(JsonKey.DESIGNATION, "", JsonKey.GROUP, "") : professionalDetails);
```

Comment:
Adding empty value is bad idea... let's not to add professionalDetails in it is empty...